Add a recipe for useful authoring-focused accessibility tools

thegbomb opened the pull request:
To add some functionality to assist with accessibility I am adding some modules to starshot and making some basic settings changes to improve things.

This is a WIP to share code and progress things to a stable state

phenaproxima commented on recipes/starshot_accessibility/recipe.yml:
```diff
@@ -0,0 +1,24 @@
+      simple_config_update:
+        high_contrast_mode: true
+        accent_color: ''
+        focus_color: ''
```
I'd actually suggest moving these Gin settings directly into `starshot_admin_theme`, and dropping it from the `recipes` list here. If these are better for accessibility, I see no reason for them not to be the defaults.

thegbomb replied:
Moved to admin_theme recipe

phenaproxima commented on recipes/starshot_accessibility/recipe.yml:
```diff
@@ -0,0 +1,24 @@
+recipes:
+  - starshot_admin_theme
+install:
+  - a11y_form_helpers
```
Not sure we should add this. It's only got 242 installs, and it's not clear from the project page what exactly it _does_, beyond a vague "makes forms more accessible". I think this is something we should consider adding separately (i.e., new issue).

thegbomb replied:
I was using a11y_form_helpers as it came from the recommendation on this page:
https://www.drupal.org/docs/getting-started/accessibility/contributed-modules-for-extending-accessibility-in-drupal#s-core-improvements
I only actually used it to ensure inline_form_errors was switched on, which I have now included in the starshot recipe itself.

phenaproxima commented on recipes/starshot_accessibility/recipe.yml:
```diff
@@ -0,0 +1,24 @@
+install:
+  - a11y_form_helpers
+  - block_aria_landmark_roles
+  - editoria11y
```
Same here -- there's a separate issue already for adding this module (#82), so if it's not strictly necessary for this recipe, maybe it'd be best to remove it.

thegbomb replied:
I added a new recipe for starshot_editoria11y but left it as optional like the multilingual. I can remove it and add this in another MR if necessary.

thegbomb commented on recipes/starshot_accessibility/recipe.yml:
```diff
@@ -0,0 +1,24 @@
+    editoria11y.settings:
+      simple_config_update:
+        ignore_elements: '.leaflet-map-pane *'
+    block.block.olivero_welcome_to_starshot:
```
The block config yml file is in recipes/starshot/config and just importing on installation, but I'll remove the tweak here adding a role of complimentary was more about showing off that it can be done, rather than it needing to be done. I'll double-check on the value of it and add it back if it is a more solid requirement.

thegbomb commented:
I have removed other changes which can be dealt with elsewhere, but have left just the editoria11y module being added in this PR.
TBD what happens to it elsewhere.
I have added it to project_browser, but it will not get installed by default.

phenaproxima commented on README.md:
```diff
@@ -65,6 +65,7 @@ We don't _quite_ support this yet, but you'll also be able to use Starshot's com
 * [Diff](https://drupal.org/project/diff)
 * [Easy Breadcrumb](https://drupal.org/project/easy_breadcrumb)
 * [ECA (Event - Condition - Action)](https://drupal.org/project/eca)
+* [editoria11y](https://www.drupal.org/project/editoria11y)
```
Let's be a little more "correct" about this.

```suggestion
* [Editoria11y Accessibility Checker](https://www.drupal.org/project/editoria11y)
```

phenaproxima commented on recipes/starshot_editoria11y/recipe.yml:
```diff
@@ -0,0 +1,10 @@
+name: Editoria11y support
```
I think that we want a little bit more generic of a name here. I propose `starshot_accessibility_tools`  as the machine name instead, to match the proposed track name in https://www.drupal.org/project/starshot/issues/3454529. It also gives us a little wiggle room to add _other_ tools here that aren't Editoria11y. (And the human-readable name should be "Accessibility tools" or similar.)

phenaproxima commented on recipes/starshot_editoria11y/recipe.yml:
```diff
@@ -0,0 +1,10 @@
+name: Editoria11y support
+type: Starshot
+description: Installs and configures editoria11y module.
```
This should be a little more generic.

```suggestion
description: Installs and configures tools to help authors make their content accessible.
```

phenaproxima commented on recipes/starshot_editoria11y/recipe.yml:
```diff
@@ -0,0 +1,10 @@
+  actions:
+    editoria11y.settings:
+      simple_config_update:
+        ignore_elements: '.leaflet-map-pane *'
```
This could use a comment to explain why this element is ignored.

phenaproxima reviewed (changes requested):
I think this looks pretty good but mostly could benefit from a rename.

phenaproxima commented on recipes/starshot_editoria11y/recipe.yml:
```diff
@@ -0,0 +1,10 @@
+type: Starshot
+description: Installs and configures editoria11y module.
+install:
+  - editoria11y
```
If `editoria11y` exposes any permissions, we probably want to update the main `starshot` recipe to ensure those permissions are added to the relevant roles (probably any role that can create or edit content should be able to use it).

thegbomb replied:
It looks like the default config for editoria11y permissions is to grant "View Editoria11y checker" and "Mark as hidden in Editoria11y"  to the authenticated user, which then includes the Content editor role. This could be okay for the site at the moment unless there is a plan to allow non-editors to create accounts.
The other permissions exposed are only granted to the administrator role.
Will recipes allow us to remove permission from a role? If so, I could remove the permissions from the Authenticated role and add it to Content editor.

phenaproxima replied:
> It looks like the default config for editoria11y permissions is to grant "View Editoria11y checker" and "Mark as hidden in Editoria11y" to the authenticated user

If that's the case, then that's probably okay for now, but can you link me to the spot in Editoria11y's code where it makes this decision?

thegbomb replied:
From what I can see, it looks like it is set here:
https://git.drupalcode.org/project/editoria11y/-/blob/2.1.x/editoria11y.install?ref_type=heads#L20

phenaproxima commented:
One typo suggestion, but otherwise I'm good with this. Nice work. I would like to consult with @timplunkett before merging just so that I can be sure that landing this will align with the defined work tracks for Starshot, and grease the wheels for further contribution.

phenaproxima commented:
I personally have no objection to this, the recipe looks good and ready to me, but I think @pameeela and @timplunkett (from the leadership team) are unsure about adding additional modules to the prototype, even if they are likely to be in the final product.

I'm assigning this to @pameeela to make the final call.

pameeela commented:
Happy to merge so folks can play around with it easily, I think it's clear that something being in the prototype doesn't imply it will be in the product and @thegbomb is across this. In this case, it's not just adding a module for the sake of it, it's actually providing something that it could be good to get feedback on.
